File: src/main/java/com/mechanitis/demo/boilerplate/example2/Conditional.java
package com.mechanitis.demo.boilerplate.example2;

@SuppressWarnings("unused")
public class Conditional {

    PortType getPortType(final String portInputValue) {

        int port = Integer.valueOf(portInputValue);
        PortType type = PortType.UNKNOWN;
        switch (port) {
            case 20:
                type = PortType.FTP;
                break;
            case 80:
                type = PortType.HTTP;
                break;
            case 8080:
                type = PortType.HTTP;
                break;
            case 27017:
                type = PortType.DATABASE;
                break;
            default:
                if (port >= 20_001 && port <= 30_000) {
                    type = PortType.SAFE;
                } else if (port >= 9080 && port <= 9092) {
                    type = PortType.BUSY;
                }
        }

        return type;
    }

    public enum PortType {
        HTTP, DATABASE, UNUSED, UNKNOWN, FTP, BUSY, SAFE;
    }
}

File: src/main/java/com/mechanitis/demo/boilerplate/example2/Conditional.kt
@file:Suppress("UNUSED_VARIABLE", "unused")

package com.mechanitis.demo.boilerplate.example2

import com.mechanitis.demo.boilerplate.example2.Conditional.PortType

fun conditional(value: Any) {

    val actual = when (value) {
        1 -> {
            println("It was one")
            20
        }
        2 -> {
            println("It was two")
            30
        }
        3, 4, 5 -> {
            println("One of these")
            35
        }
        is Int -> {
            println("It's an Int")
            20
        }
        in 0..100 -> {
            println("It was somewhere in between these numbers")
            40
        }
        else -> {
            0
        }
    }

}

class PortValidator {

    fun getPortType(port: Any): PortType {

        val type = when (port) {
            20 -> PortType.FTP
            80, 8080 -> PortType.HTTP
            27017 -> PortType.DATABASE
            in 20001..30000 -> PortType.SAFE
            in 9080..9092 -> PortType.BUSY
            !is Int -> throw RuntimeException("Not a valid port number")
            else -> PortType.UNKNOWN
        }

        return type
    }
}


